src/components/Polls/CreatePoll.vue

```diff
@@ -0,0 +1,75 @@
+<template>
+  <p class="text-gray-600 mb-4">
+    Submit responses and vote on them anonymously. When a response gets enough votes, voters for that response reveal their
+    names.
+  </p>
+  <div class="mb-4">
+    <span class="font-semibold">Poll title:</span>
+    <div>
+      <input v-model="title" class="border rounded border-gray-500 w-full h-12 px-3" type="text" />
+    </div>
+  </div>
+  <div class="mb-4">
+    <span class="font-semibold">Description:</span>
+    <div>
+      <textarea v-model="description" rows="2" class="h-full border rounded border-gray-500 w-full p-3" type="text">
+      </textarea>
+    </div>
+  </div>
+  <div class="mb-4">
+    <p class="font-semibold">End date:</p>
+
+    <!-- Date picker -->
+    <div class="flex max-w-sm border rounded border-gray-500 w-full">
+      <input v-model="expirationDate" class="rounded w-full p-3" type="date" :min="today" />
+    </div>
+  </div>
+  <!-- Error message -->
+  <div class="text-harvard-red mt-2">{{ message }}</div>
+</template>
+
+<script setup>
+import { ref } from '@vue/reactivity'
+import useStore from '../../composables/global/useStore'
+import { useRoute } from 'vue-router'
+
+const { createPoll } = useStore
+const title = ref('')
+const description = ref('')
+const message = ref('')
+const route = useRoute()
+
+const today = new Date().toISOString().split('T')[0]
+const date = new Date()
+date.setDate(date.getDate() + 5)
+const defaultDate = date.toISOString().split('T')[0]
+
+const expirationDate = ref(defaultDate)
+
+defineExpose({
+  processCreate
+})
+
+const emit = defineEmits(['createSuccess'])
+
+function processCreate() {
+  if (!hasTitle()) {
+    message.value = 'A title is required.'
+    return
+  }
+  const expirationDateValue = new Date(expirationDate.value)
+  createPoll({
+    title: title.value,
+    description: description.value,
+    expirationDate: expirationDateValue,
+    topic: route.params.channelId,
+    slug: 'TODO'
```

Comment:
On slug, I was just trying to be parallel to threads, but it should not be something that needs to be set by the FE. I will either make it optional, remove it, or set it by default on the BE. Do you see some use for it? Would you like to have URLs that reference the topic rather than just the poll ID, for example?

Reply:
Like if you wanted the title of the poll to be used as the URL, or part of the URL for that particular poll.

Reply:
Confirming you do not need to pass slug: it will get automatically created for new polls. This is already the built in behavior.